### ISSUE TITLE
Vars should also be kept per task

### ISSUE THREAD

**Issue by jhoekx** [ansible/ansible#111]
This currently fails:

``` yaml
- hosts: rh5man
  tasks:
  - include: networking.yml interface="eth0" mac="52:54:00:38:CC:5A" ipaddr="192.168.122.35" netmask="255.255.255.0"
  - include: networking.yml interface="eth1" mac="12:34:56:78:90:12" ipaddr="127.0.0.2" netmask="255.255.255.255"
```

With networking.yml:

``` yaml

---
- name: update network interface {{interface}}
  action: template src=res/ifcfg dest=/etc/sysconfig/network-scripts/ifcfg-{{ interface }}
```

It results in:

``` bash
[root@rh5man ~]# cat /etc/sysconfig/network-scripts/ifcfg-eth0
DEVICE=eth1
BOOTPROTO=static
DHCPCLASS=
HWADDR=12:34:56:78:90:12
IPADDR=127.0.0.2
NETMASK=255.255.255.255
ONBOOT=yes
```

With ifcfg-eth1 identical.

This is because the vars are stored per play. In this case you need vars per task.

Note that substitution in the playbook is correct because the substitutions for eth0 are made before the variables of eth1 are read.

**Comment by mpdehaan** [comment]
actually I'm not sure you need vars per task, because the first include can/should pass on those vars to the first instance of the include and then the second would get different vars.

What it seems like you're saying is the vars to the includes are made available to templates and not the action statements

I think we are templating the include files (or that was the plan), so I'm a bit confused but I haven't quite debugged it yet.

If you want to see why the interface variable isn't being passed down right with the include, I think that will solve the problem... does that make sense?

**Comment by jhoekx** [comment]
I will try to explain it better.

Templating the include file is working OK, with the correct variables. It's the values used in the templates that are wrong. In my example above, both ifcfg-eth0 and ifcfg-eth1 will be identical.

The variables passed on to the include are 'leaked' to the play scope.

This is the code responsible for handling the includes:

``` python
        inject_vars = self._get_vars(play, dirname)
        for i,x in enumerate(include_tokens):
            if x.find("=") != -1:
                (k,v) = x.split("=")
                inject_vars[k] = v
```

`inject_vars` is per play. If variables have the same name, they will be overwritten. For use in the playbook includes, that does not matter. The first included file is fully templated before the variables of the second one are read. For remote execution, it does matter. Setup transfers the variables to /etc/ansible/setup. This happens after the whole play has been processed.

A complete, simple example:

The main playbook:

``` yaml
---
- hosts: rh5man
  tasks:
  - include: hello.yml nr=1 value="worldA"
  - include: hello.yml nr=2 value="worldB"
```

hello.yml:

``` yaml
---
- name: Print hello {{ nr }}
  action: template src=res/hello.txt dest=/root/hello-{{ nr }}.txt
```

hello.txt:

```
{{ value }}

```

The result:

``` bash
[root@rh5man ~]# tail hello-*
==> hello-1.txt <==
worldB
==> hello-2.txt <==
worldB
```

The expected result:

``` bash
[root@rh5man ~]# tail hello-*
==> hello-1.txt <==
worldA
==> hello-2.txt <==
worldB
```

**Comment by mpdehaan** [comment]
"The variables passed on to the include are 'leaked' to the play scope."

Ok, this makes perfect sense.   Thanks ... Looks like the code needs to make a copy of the hash inject_vars prior to the modifications to inject_vars and then make the template call using the copy.

Should be an easy fix.   Will probably have a chance to look at this tonight unless you want to beat me to it.

**Comment by mpdehaan** [comment]
fixed per merging of #114
